bin/utils/subkey/src/rpc.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Helper to run commands against current node RPC
+
+use futures::Future;
+use hyper::rt;
+use node_primitives::Hash;
+use sc_rpc::author::AuthorClient;
+use jsonrpc_core_client::transports::http;
+use sp_core::Bytes;
+
+pub struct RpcClient { url: String }
+
+impl RpcClient {
+	pub fn new(url: String) -> Self { Self { url } }
+
+	pub fn insert_key(
+		&self,
+		key_type: String,
+		suri: String,
+		public: Bytes,
+	) {
+		let url = self.url.to_string();
+
+		rt::run(rt::lazy(move || {
```

Comment:
Do we really need `hyper` for that? Couldn't we just spawn on any runtime?

Reply:
I'd suggest there is little use of worrying about it, since those are just synchronous requests initiated from cli, and reusing some runtime external will just complicate everything?

Reply:
Chatted off github